sieves/tasks/predictive/ner/core.py

```diff
@@ -0,0 +1,221 @@
+    | glix_.Result
+    | huggingface_.Result
+    | instructor_.Result
+    | langchain_.Result
+    | ollama_.Result
+    | outlines_.Result
+)
+# Define a proper TaskBridge that includes all the specific NER bridges
+_TaskBridge: TypeAlias = DSPyNER | GliXNER | InstructorNER | LangChainNER | OllamaNER | OutlinesNER
+
+
+class Entity(pydantic.BaseModel):
+    text: str
+    context: str
+    entity_type: str
+
+
+class TaskFewshotExample(pydantic.BaseModel):
+    text: str
+    entities: list[Entity]
+
+
+class NER(PredictiveTask[_TaskPromptSignature, _TaskResult, _TaskBridge]):
+    def __init__(
+        self,
+        entities: list[str],
+        engine: Engine[EnginePromptSignature, EngineResult, EngineModel, EngineInferenceMode],
+        task_id: str | None = None,
+        show_progress: bool = True,
+        include_meta: bool = True,
+        prompt_template: str | None = None,
+        prompt_signature_desc: str | None = None,
+        fewshot_examples: Iterable[TaskFewshotExample] = (),
+    ) -> None:
+        """ "
+        Initializes new PredictiveTask.
+        :param task_id: Task ID.
+        :param show_progress: Whether to show progress bar for processed documents.
+        :param include_meta: Whether to include meta information generated by the task.
+        :param prompt_template: Custom prompt template. If None, task's default template is being used.
+        :param prompt_signature_desc: Custom prompt signature description. If None, default will be used.
+        :param fewshot_examples: Few-shot examples.
+        """
+        self._entities = entities or ["PERSON", "LOCATION", "ORGANIZATION"]
+        super().__init__(
+            engine=engine,
+            task_id=task_id,
+            show_progress=show_progress,
+            include_meta=include_meta,
+            overwrite=False,
+            prompt_template=prompt_template,
+            prompt_signature_desc=prompt_signature_desc,
+            fewshot_examples=fewshot_examples,
+        )
+        self._fewshot_examples: Iterable[TaskFewshotExample]
+
+    def _init_bridge(self, engine_type: EngineType) -> _TaskBridge:
+        """Initialize bridge.
+        :return: Engine task.
+        :raises ValueError: If engine type is not supported.
+        """
+        bridge_types = {
+            EngineType.langchain: LangChainNER,
+            EngineType.ollama: OllamaNER,
+            EngineType.outlines: OutlinesNER,
+            EngineType.dspy: DSPyNER,
+            EngineType.instructor: InstructorNER,
+            EngineType.glix: GliXNER,
+        }
+        try:
+            bridge_class = bridge_types[engine_type]
+            result = bridge_class(
+                task_id=self._task_id,
+                prompt_template=self._custom_prompt_template,
+                prompt_signature_desc=self._custom_prompt_signature_desc,
+                entities=self._entities,
+            )
+            # Cast the result to the appropriate type to satisfy mypy
```

Comment:
Check other tasks to see how to resolve this `mypy` error. If you want to temporarily disable mypy errors, use the corresponding `type: ignore[...]` statement in that line (but use the concrete error code, not just a general ignore).